ui/dashboards/src/components/DashboardToolbar/DashboardToolbar.tsx

```diff
@@ -61,30 +61,54 @@ export const DashboardToolbar = (props: DashboardToolbarProps) => {
 
   return (
     <>
-      {isEditMode ? (
-        <Stack spacing={1} data-testid={testId}>
-          <Box px={2} py={1.5} display="flex" sx={{ backgroundColor: (theme) => theme.palette.primary.main + '30' }}>
-            {dashboardTitle}
+      <Stack data-testid={testId}>
+        <Box
+          px={2}
+          py={1.5}
+          display="flex"
+          sx={{ backgroundColor: (theme) => theme.palette.primary.main + (isEditMode ? '30' : '0') }}
+        >
+          {dashboardTitle}
+          {isEditMode ? (
             <Stack direction="row" gap={1} ml="auto">
               {isReadonly && (
                 <Alert severity={'warning'} sx={{ backgroundColor: 'transparent', padding: 0 }}>
                   Dashboard managed via code only. Download JSON and commit changes to save.
                 </Alert>
               )}
+              <Stack direction="row" spacing={0.5} ml={1} whiteSpace="nowrap">
+                <EditVariablesButton />
+                <EditDatasourcesButton />
+                <AddPanelButton />
+                <AddGroupButton />
+              </Stack>
               <SaveDashboardButton onSave={onSave} isDisabled={isReadonly} />
               <Button variant="outlined" onClick={onCancelButtonClick}>
                 Cancel
               </Button>
             </Stack>
-          </Box>
-          <Box
-            sx={{
-              display: 'flex',
-              width: '100%',
-              alignItems: 'start',
-              padding: (theme) => theme.spacing(1, 2, 0, 2),
-            }}
-          >
+          ) : (
+            <>
+              {isBiggerThanSm && (
+                <Stack direction="row" gap={1} ml="auto">
+                  <EditButton onClick={onEditButtonClick} />
+                </Stack>
+              )}
+            </>
+          )}
+        </Box>
+        <Box
+          sx={{
+            display: 'flex',
+            width: '100%',
+            alignItems: 'start',
+            padding: (theme) => theme.spacing(1, 2, 0, 2),
+            flexDirection: isBiggerThanMd ? 'row' : 'column',
+            flexWrap: 'nowrap',
+            gap: 1,
+          }}
+        >
+          <Box flexGrow={1} flexShrink={1}>
```

Comment:
```suggestion
          <Box flexGrow={1} flexShrink={1} width="100%">
```
Will fix: 
![image](https://github.com/perses/perses/assets/5657041/81109c85-8199-438d-88af-061c9441c258)

Now:
![image](https://github.com/perses/perses/assets/5657041/38038c52-17ca-4fdd-967d-e6cb50f57fb9)